File: Day3/DSA/Group_Anagrams.java
Question :- https://leetcode.com/problems/group-anagrams/description/

Code:- 

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        HashMap<String, List<String>> map = new HashMap<String, List<String>>();

        for (String i : strs) {
            char arr[] = i.toCharArray();
            Arrays.sort(arr);
            String str = new String(arr);

            if (!map.containsKey(str)) {
                map.put(str, new ArrayList<>());
            }

            map.get(str).add(i);
        }

        return new ArrayList<>(map.values());
    }
}

File: Day3/DSA/Group_Anagrams.cpp

Question : -https : // leetcode.com/problems/group-anagrams/description/

                    
class Solution
{
public:
    vector<vector<string>> groupAnagrams(vector<string> &strs)
    {
        unordered_map<string, vector<string>> mp;

        for (auto i : strs)
        {
            string str = i;
            sort(str.begin(), str.end());
            mp[str].push_back(i);
        }

        vector<vector<string>> v;
        for (auto i : mp)
        {
            v.push_back(i.second);
        }

        return v;
    }
};
